release-notes.txt

```diff
@@ -1,3 +1,363 @@
+RIOT-2019.01 - Release Notes
+============================
+RIOT is a multi-threading operating system which enables soft real-time
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2019.01 release includes a number of new features including DMA support on
+all STM32 boards, WiFi with IP connectivity on ESP32 and ESP8266, and riotboot,
+a simple bootloader supporting multiple firmware slots, compatible with most Arm
+Cortex-M boards supported by RIOT. Support for several new boards and new sensors
+was added, as well as for the new CPU architecture Armv8-M (Cortex-M23). Last
+but not least, this release contains a number of bug fixes and enhances test
+automation.
+
+About 387 pull requests with about 912 commits have been merged since the last
+release and about 36 issues have been solved. 54 people contributed with code
+in 92 days. Approximately 1525 files have been touched with 171769 insertions
+and 46121 deletions.
```

Comment:
Are these numbers recent or just copy-pasted?

Reply:
a quick check here shows more like 425 PRs merged, and 1285 commits since the 2018.10 release (on Nov. 13th, right?)

I used the below methods, taking:
- for the merged PR count, [this](https://api.github.com/search/issues?q=repo:RIOT-OS/RIOT+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-11-13&sort=mergedOS+is:merged+sort:author-date-asc+merged:%3E2018-10-01&sort=merged) github API call
- for the number of commits, the below git command
`$ git rev-list --after="2018-11-13T00:00" --count master
`

Reply:
912 commits is right according to

```sh
./dist/tools/release-stats/release-stats.sh 2019.01-devel 2019.01-RC3
```

But the rest is off a bit. The number of issues is [also too small](https://github.com/RIOT-OS/RIOT/issues?q=is%3Aissue+closed%3A%3E%3D2018-11-13+is%3Aclosed) and we got [429 PRs merged](https://github.com/RIOT-OS/RIOT/pulls?utf8=%E2%9C%93&q=is%3Apr+merged%3A%3E%3D2018-11-13+is%3Amerged)

Reply:
Sorry, just saw your comment now ^^"

Reply:
The results of the release stats script were

```html
    <tr>
        <td>2019.01-RC3</td>
        <td>7131</td>
        <td>2010903 (1914102)</td>
        <td>110 days</td>
        <td>912</td>
        <td>1554 files changed, 176241 insertions(+), 50593 deletions(-)</td>
        <td>199 Martine Lenders<br />
            128 Alexandre Abadie<br />
            112 Gunar Schorcht<br />
            48 Gaëtan Harter<br />
            48 Kees Bakker<br /></td>
    </tr>
```

btw. They are meant for [this table](https://github.com/RIOT-OS/RIOT/wiki/release-statistics). So fourth cell is the development time, fifth the number of commits, and sixth the diff stats.

Number of contributors can be counted with 

```
$ git shortlog --no-merges -n -s "2019.01-devel".."2019.01-RC3" | wc -l
54
```

Reply:
>     git rev-list --after="2018-11-13T00:00" --count master

This method also includes things that are already merged in master but not backported to the release branch. To get the number of PRs I used the PR listed in GitHub but filtered on the Milestone 2019.01. I tried to include everything I could but maybe some were missed.

> The number of issues is [also too small](https://github.com/RIOT-OS/RIOT/issues?q=is%3Aissue+closed%3A%3E%3D2018-11-13+is%3Aclosed)

because it only includes the issues labelled as bug.

Reply:
About the number of files touched and global changeset, I did:

```
$ git diff --shortstat 2019.01-devel..2019.01-RC3
 1525 files changed, 171769 insertions(+), 46121 deletions(-)
```

Reply:
I updated the number of days. Do we agree with the other numbers ?

Reply:
About `git diffstats`, `dist/tools/release-stats/release-stats.sh` uses the `--no-renames` option.  IIUC, with this, the statistics also includes files that were just renamed. Why not, so I'll update the values.